explorer/interpreter/resolve_unformed.cpp

```diff
@@ -0,0 +1,234 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "explorer/interpreter/resolve_unformed.h"
+
+#include <unordered_map>
+
+#include "common/check.h"
+#include "explorer/ast/ast.h"
+#include "explorer/ast/expression.h"
+#include "explorer/ast/pattern.h"
+#include "explorer/ast/static_scope.h"
+#include "explorer/common/error_builders.h"
+#include "explorer/common/nonnull.h"
+
+using llvm::cast;
+
+namespace Carbon {
+
+// Aggregate information about a ValueNodeView being analyzed.
+struct FlowFact {
+  bool is_formed;
```

Comment:
Is this assuming it always knows or doesn't know? What about something like:

```
fn Init(x: i32*) {
  *x = 0;
}

fn Main() -> i32 {
  var x: i32;
  var f: auto = Init;
  f(&x);
  return x;
}
```

https://carbon.compiler-explorer.com/z/EM5h79afa

Reading the code, it seems like this is uncertain: maybe "may_be_formed" would be a better name?

Reply:
Renamed `is_formed` as `may_be_formed`.